python_multipart/multipart.py

```diff
@@ -1656,7 +1678,7 @@ def on_header_value(data: bytes, start: int, end: int) -> None:
                 header_value.append(data[start:end])
 
             def on_header_end() -> None:
-                headers[b"".join(header_name)] = b"".join(header_value)
+                headers[b"".join(header_name).decode().lower()] = b"".join(header_value)
```

Comment:
Do we need to decode?

Reply:
I think the headers dict is easier to use if the keys are strings rather than bytes, so I used decode to do the transformation. The string is guaranteed to be valid since it can only contain a subset of ASCII characters (it's filtered as it's built). Can be bytes if you prefer, in which case decode isn't needed.

Reply:
Does it need to be in this PR?